tests/device_tests/ethereum/test_sign_verify_message.py

```diff
@@ -0,0 +1,58 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2019 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib import ethereum
+from trezorlib.tools import parse_path
+
+from ...common import parametrize_using_common_fixtures
+
+pytestmark = [pytest.mark.altcoin, pytest.mark.ethereum]
+
+
+@parametrize_using_common_fixtures(
+    "ethereum/signmessage.json",
+)
+def test_signmessage(client, parameters, result):
+    res = ethereum.sign_message(
+        client, parse_path(parameters["path"]), parameters["msg"]
+    )
+    assert res.address == parameters["address"]
+    assert res.signature.hex() == result["sig"]
+
+
+@parametrize_using_common_fixtures(
+    "ethereum/verifymessage.json",
+)
+def test_verify(client, parameters, result):
+    res = ethereum.verify_message(
+        client, parameters["address"], bytes.fromhex(result["sig"]), parameters["msg"]
+    )
+    assert res is True
+
+
+def test_verify_invalid(client):
```

Comment:
let's modify this so that it's clearer what is going on

1. pick one of the vectors in JSON, use the data here
2. call `verify_message` on the unmodified vector, assert result is `True`
3. change one byte of `signature`, call `verify_message`, assert result is `False`
4. change the message but use the original signature, call `verify_message`, assert false

Reply:
> * pick one of the vectors in JSON, use the data here

pick the first vector and say in comment that it's the first vector from JSON

Reply:
Modified in 59c7eb7

Reply:
please also add the second failing condition: same signature, changed message

Reply:
(both are important. first checks that there exists a signature that can fail, the second checks that the signature depends on the message)

Reply:
What about also changing the address?

Reply:
yes, that's also a good idea

Reply:
Message-changed-test-case in 0fdabfa